src/main/scala/mesosphere/marathon/core/storage/MarathonState.scala

```diff
@@ -0,0 +1,22 @@
+package mesosphere.marathon.core.storage
+
+import akka.util.ByteString
+import com.google.protobuf.{ MessageLite, Parser }
+import mesosphere.marathon.state.Timestamp
+
+trait MarathonState[Proto <: MessageLite] {
```

Comment:
I see it the same way as a separate Unmarshaller/Marshaller. 
Even if we implement it directly (as it is now) we should not force this.